clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -368,7 +373,11 @@ public void onSuccess(Void value) {
 
             @Override
             public void onFailure(RuntimeException e) {
-                cb.onComplete(offsets, e);
+                if (e instanceof RetriableException) {
+                    cb.onComplete(offsets, RETRIABLE_COMMIT_EXCEPTION);
+                } else {
```

Comment:
Seems like the exception could already be an instance of `CommitFailedException`.